### PR TITLE
Refactor/upgrade eslint and friends

### PR DESCRIPTION
**Pull request by areisle**
relates to #20

* updates eslint and eslint plugins to latest version
* moves linting from build to run as a separate task just before tests
* fixes or disables the rule all the new violations from updating

### CODE REVIEW

File: src/components/Auth/index.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ const useAuth = () => {
   return state;
 };
 
-const AuthProvider = (props) => {
+function AuthProvider(props) {
```
Comment (creisle) [inline src/components/Auth/index.js:41]: why did you change these to function?

File: src/components/RecordAutocomplete/index.js [hunk 2]
```diff
@@ -248,10 +248,10 @@ const RecordAutocomplete = (props) => {
           InputProps={{
             ...params.InputProps,
             endAdornment: (
-              <React.Fragment>
+              <>
```
Comment (creisle) [inline src/components/RecordAutocomplete/index.js:251]: These were set to React.Fragment for readability

**Review by creisle — COMMENTED** [review]
The diff on this PR is very large. It would be a lot easier to review if the auto-fix style changes were separated from the upgrades and done in seperate PRs for each (if large) of the preferred linting rule changes

**Comment by areisle** [comment]
> The diff on this PR is very large. It would be a lot easier to review if the auto-fix style changes were separated from the upgrades and done in separate PRs for each (if large) of the preferred linting rule changes

I did separate all the auto-fixed styles into their own commit, but I didn't make this multiple PRs since, in order for it to have been multiple PRs, I would have either had to have either:
* removed eslint from build but not add it to actions so it wouldn't be running at all
* allow a failing build and still merge (not sure that's possible with the settings for this repo anyway)
* or disabled any rule that changed/added in latest airbnb, react etc and then reverse all that in the next PR

none of which seemed like particularly good options

File: src/components/Auth/index.js (hunk 1 above)
Comment (areisle) [reply src/components/Auth/index.js:41]: this was autofixed to conform with [this rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md) from the latest version of the react eslint plugin

File: src/components/RecordAutocomplete/index.js (hunk 2 above)
Comment (areisle) [reply src/components/RecordAutocomplete/index.js:251]: this was autofixed to conform with the latest version of the eslint react plugin due to [this rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-fragments.md)

File: src/components/FormLayout/FieldGroup.js [hunk 3]
```diff
@@ -113,11 +113,8 @@ const FieldGroup = ({
       fields.push(wrapper);
     }
   });
-  return (
-    <React.Fragment>{fields}</React.Fragment>
-  );
-};
-
+  return fields;
```
Comment (kttkjl) [inline src/components/FormLayout/FieldGroup.js:116]: Would it be beneficial for [ `allowedExpressions` ](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-useless-fragment.md#allowexpressions) be turned on for this case? since I am assuming that the project will be ported over to typescript eventually? If not at least the rule being there would not remove the indication that this return was a react fragment

File: src/views/GraphView/components/GraphComponent/kbgraph.js [hunk 4]
```diff
@@ -107,7 +108,7 @@ class PropsMap {
   }
 
   /**
-   * Loads a node's properties into the propsmap.
+   * Loads a node's properties into the proposal.
```
[the remark this answers is not in the document]
Comment (areisle) [reply src/views/GraphView/components/GraphComponent/kbgraph.js:111]: huh, not sure how that happened :thinking:

File: src/components/FormLayout/FieldGroup.js (hunk 3 above)
Comment (areisle) [reply src/components/FormLayout/FieldGroup.js:116]: I'm fine with turning that on or disabling the rule or waiting and adding back fragments where necessary once we do get to typescript. mostly I was trying to add as little extra customization to the eslint config as possible since that makes things easier with many projects

**Comment by creisle** [comment]
I'm am fine with these changes in principle but due to the size of the PR I won't have time to review this one so I am removing myself as reviewer. @kttkjl can you take a look?

**Comment by areisle** [comment]
closed in favour of https://github.com/bcgsc/pori_graphkb_client/pull/43
